<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from j2s7s300_standalone.xacro      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<!-- j2s7s300 refers to jaco v2 7DOF spherical 3fingers -->
<robot name="j2s7s300" xmlns:body="http://playerstage.sourceforge.net/gazebo/xmlschema/#body" xmlns:controller="http://playerstage.sourceforge.net/gazebo/xmlschema/#controller" xmlns:gazebo="http://playerstage.sourceforge.net/gazebo/xmlschema/#gz" xmlns:geom="http://playerstage.sourceforge.net/gazebo/xmlschema/#geom" xmlns:interface="http://playerstage.sourceforge.net/gazebo/xmlschema/#interface" xmlns:joint="http://playerstage.sourceforge.net/gazebo/xmlschema/#joint" xmlns:model="http://playerstage.sourceforge.net/gazebo/xmlschema/#model" xmlns:physics="http://playerstage.sourceforge.net/gazebo/xmlschema/#physics" xmlns:renderable="http://playerstage.sourceforge.net/gazebo/xmlschema/#renderable" xmlns:rendering="http://playerstage.sourceforge.net/gazebo/xmlschema/#rendering" xmlns:sensor="http://playerstage.sourceforge.net/gazebo/xmlschema/#sensor" xmlns:xacro="http://ros.org/wiki/xacro" xmlns:xi="http://www.w3.org/2001/XInclude">
  <link name="j2s7s300_link_finger_1">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
      <material name="fingerLinkMaterial">
      <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/> 
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_1" type="revolute">
    <parent link="j2s7s300_link_7"/>
    <child link="j2s7s300_link_finger_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.570796327 .649262481663582 1.35961148639407" xyz="0.00279 0.03126 -0.11467"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
  </joint>
  <link name="j2s7s300_link_finger_tip_1">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
      <material name="fingerTipMaterial">
        <color rgba="0.2 0.2 0.2 1"/>     
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_tip_1" type="fixed">
    <parent link="j2s7s300_link_finger_1"/>
    <child link="j2s7s300_link_finger_tip_1"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044 -0.003 0"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
    <!--<mimic joint="0.698131700798" multiplier="1.0" offset="0.0"/>-->
  </joint>
  <link name="j2s7s300_link_finger_2">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
      <material name="fingerLinkMaterial">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/> 
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_2" type="revolute">
    <parent link="j2s7s300_link_7"/>
    <child link="j2s7s300_link_finger_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.570796327 .649262481663582 -1.75545216211587" xyz="-0.02226 -0.02707 -0.11482"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
  </joint>
  <link name="j2s7s300_link_finger_tip_2">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
      <material name="fingerTipMaterial">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_tip_2" type="fixed">
    <parent link="j2s7s300_link_finger_2"/>
    <child link="j2s7s300_link_finger_tip_2"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044 -0.003 0"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
    <!-- <mimic joint="0.698131700798" multiplier="1.0" offset="0.0"/> -->
  </joint>
  <link name="j2s7s300_link_finger_3">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
      <material name="fingerLinkMaterial">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_proximal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_3" type="revolute">
    <parent link="j2s7s300_link_7"/>
    <child link="j2s7s300_link_finger_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="-1.570796327 .649262481663582 -1.38614049188413" xyz="0.02226 -0.02707 -0.11482"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
  </joint>
  <link name="j2s7s300_link_finger_tip_3">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
      <material name="fingerTipMaterial">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/finger_distal.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_finger_tip_3" type="fixed">
    <parent link="j2s7s300_link_finger_3"/>
    <child link="j2s7s300_link_finger_tip_3"/>
    <axis xyz="0 0 1"/>
    <origin rpy="0 0 0" xyz="0.044 -0.003 0"/>
    <limit effort="30" lower="0" upper="2" velocity="5"/>
    <!-- <mimic joint="0.698131700798" multiplier="1.0" offset="0.0"/> -->
  </joint>
  <link name="root">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
      <!--<material name="Black" /> -->
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
  </link>
  <link name="j2s7s300_link_base">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/base.dae"/>
      </geometry>
      <material name="baseMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/base.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_base" type="fixed">
    <parent link="root"/>
    <child link="j2s7s300_link_base"/>
    <axis xyz="0 0 0"/>
    <limit effort="30" lower="0" upper="0" velocity="0.35"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
  </joint>
  <link name="j2s7s300_link_1">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/shoulder.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/> 
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/shoulder.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_1" type="continuous">
    <parent link="j2s7s300_link_base"/>
    <child link="j2s7s300_link_1"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="0.35"/>
    <origin rpy="0 3.14159265359 0" xyz="0 0 0.15675"/>
  </joint>
  <link name="j2s7s300_link_2">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/arm_half_1.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/arm_half_1.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_2" type="continuous">
    <parent link="j2s7s300_link_1"/>
    <child link="j2s7s300_link_2"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-2.35" upper="2.35" velocity="0.35"/>
    <origin rpy="-1.57079632679 0 3.14159265359" xyz="0 0.0016 -0.11875"/>
  </joint>
  <link name="j2s7s300_link_3">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/arm_half_2.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/arm_half_2.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_3" type="continuous">
    <parent link="j2s7s300_link_2"/>
    <child link="j2s7s300_link_3"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="0.35"/>
    <origin rpy="1.57079632679 0 3.14159265359" xyz="0 -0.205 0"/>
  </joint>
  <link name="j2s7s300_link_4">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/forearm.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/forearm.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_4" type="continuous">
    <parent link="j2s7s300_link_3"/>
    <child link="j2s7s300_link_4"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-2.65" upper="2.65" velocity="0.35"/>
    <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0 -0.205"/>
  </joint>
  <link name="j2s7s300_link_5">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/wrist_spherical_1.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/wrist_spherical_1.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_5" type="continuous">
    <parent link="j2s7s300_link_4"/>
    <child link="j2s7s300_link_5"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="0.35"/>
    <origin rpy="-1.57079632679 0 3.14159265359" xyz="0 0.2073 -0.0114"/>
  </joint>
  <link name="j2s7s300_link_6">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/wrist_spherical_2.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/wrist_spherical_2.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_6" type="continuous">
    <parent link="j2s7s300_link_5"/>
    <child link="j2s7s300_link_6"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-2.25" upper="2.25" velocity="0.35"/>
    <origin rpy="1.57079632679 0 3.14159265359" xyz="0 0 -0.10375"/>
  </joint>
  <link name="j2s7s300_link_7">
    <visual>
      <geometry>
        <mesh filename="kinova_description/meshes/hand_3finger.dae"/>
      </geometry>
      <material name="linkMaterial">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="kinova_description/meshes/hand_3finger.dae"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_joint_7" type="continuous">
    <parent link="j2s7s300_link_6"/>
    <child link="j2s7s300_link_7"/>
    <axis xyz="0 0 1"/>
    <limit effort="30" lower="-6.28318530718" upper="6.28318530718" velocity="0.35"/>
    <origin rpy="-1.57079632679 0 3.14159265359" xyz="0 0.10375 0"/>
  </joint>
  <link name="j2s7s300_end_effector">
    <visual>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </visual>
    <collision>
      <geometry>
        <box size="0 0 0"/>
      </geometry>
    </collision>
  </link>
  <joint name="j2s7s300_end_effector_offset" type="fixed">
    <parent link="j2s7s300_link_7"/>
    <child link="j2s7s300_end_effector"/>
    <axis xyz="0 0 0"/>
    <limit effort="30" lower="0" upper="0" velocity="0.35"/>
    <origin rpy="3.14159265359 0 0" xyz="0 0 -0.1600"/>
  </joint>
</robot>

